keel-intent-aws/src/main/kotlin/com/netflix/spinnaker/keel/intent/aws/loadbalancer/ClassicLoadBalancerIntentProcessor.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.keel.intent.aws.loadbalancer
+
+import com.amazonaws.services.elasticloadbalancing.model.LoadBalancerDescription
+import com.fasterxml.jackson.databind.MapperFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.collect.Maps
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
+import com.netflix.spinnaker.keel.clouddriver.CloudDriverService
+import com.netflix.spinnaker.keel.dryrun.ChangeSummary
+import com.netflix.spinnaker.keel.intent.*
+import com.netflix.spinnaker.keel.tracing.TraceRepository
+import net.logstash.logback.argument.StructuredArguments.value
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import com.netflix.awsobjectmapper.AmazonObjectMapperConfigurer
+import com.netflix.spinnaker.keel.*
+import java.util.*
+
+@Component
+class ClassicLoadBalancerIntentProcessor
+@Autowired constructor(
+  private val traceRepository: TraceRepository,
+  private val cloudDriverService: CloudDriverService,
+  private val clouddriverCache: CloudDriverCache,
+  objectMapper: ObjectMapper,
+  private val classicLoadBalancerConverter: ClassicLoadBalancerConverter
+) : IntentProcessor<LoadBalancerIntent> {
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  private val objectMapper = objectMapper.apply {
+    configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true)
+    AmazonObjectMapperConfigurer.configure(this)
+
+    propertyNamingStrategy = null
+  }
+
+  override fun supports(intent: Intent<IntentSpec>) = intent.spec is ClassicLoadBalancerSpec
+
+  override fun converge(intent: LoadBalancerIntent): ConvergeResult {
+    val changeSummary = ChangeSummary(intent.id())
+    val spec = intent.spec as ClassicLoadBalancerSpec
+
+    log.info("Converging state for {}", value("intent", intent.id()))
+
+    getLoadBalancer(spec)?.let {
+      val differences = Maps.difference(
```

Comment:
Not quite sure how best to detect differences ... I don't really want to do explicit checks against each particular key/value.

Reply:
I wrote `StateInspector` to build up diffs; I didn't know about this Guava class. StateInspector groups diffs by root-level element, although I want to have more granular control.

From an end-user's perspective, if I'm working with a loadbalancer's healthcheck, I'd just like to see the diff of the healthcheck if something as changed, not an individual property. However, if I'm in some other part, I might just want to see the individual property change. It seems contextual to me... in an ideal world I'd like to have some utility to allow developers to decide what units are more/less granular.

The fact we're generating diffs is the important bit at this point. I'd like to see what we and early adopters find most useful before getting fancy, though.